tests/drift_kinetic_push/drift_kinetic_interpolation_ex1.cpp

```diff
@@ -0,0 +1,142 @@
+#include "drift_kinetic_push.h"
+
+static constexpr char help[] =
+  "Test drift-kinetic Esirkepov interpolation: simple field interpolation.\n"
+  "Tests interpolation of arbitrary E, B, and gradB fields on a particle.\n";
+
+void get_analytical_fields(Vector3R& r, Vector3R& E_p, Vector3R& B_p, Vector3R& gradB_p)
+{
+  E_p = Vector3R(1.0, 1.0, 1.0);
+
+  B_p = Vector3R(0.0, 0.0, 1.0);
+
+  gradB_p = r;
+}
+
+PetscErrorCode initialize_grid_fields(DM da, Vec E_vec, Vec B_vec, Vec gradB_vec)
+{
+  PetscFunctionBeginUser;
+
+  Vector3R*** E_arr, ***B_arr, ***gradB_arr;
+
+  PetscCall(DMDAVecGetArrayWrite(da, E_vec, &E_arr));
+  PetscCall(DMDAVecGetArrayWrite(da, B_vec, &B_arr));
+  PetscCall(DMDAVecGetArrayWrite(da, gradB_vec, &gradB_arr));
+
+  Vector3I start, size;
+  PetscCall(DMDAGetCorners(da, REP3_A(&start), REP3_A(&size)));
+
+  for (PetscInt k = start[Z]; k < start[Z] + size[Z]; k++) {
+    for (PetscInt j = start[Y]; j < start[Y] + size[Y]; j++) {
+      for (PetscInt i = start[X]; i < start[X] + size[X]; i++) {
+        Vector3R r(i, j, k);
```

Comment:
To map cell index to cell coordinate multiply by grid size:
```suggestion
        Vector3R r(i * dx, j * dy, k * dz);
```